src/main/java/edu/ucsb/cs156/gauchoride/controllers/RideRequestController.java

```diff
@@ -0,0 +1,132 @@
+    }
+
+    @ApiOperation(value = "Get a list of all ride requests for current rider")
+    @PreAuthorize("hasRole('ROLE_RIDER')")
+    @GetMapping("/all_rider")
+    public ResponseEntity<String> rideRequestsRider(
+        @ApiParam(name="userid", type="Long", value="userid of the rider", example="123", required=true) @RequestParam Long userid
+        )
+        throws JsonProcessingException {
+
+        Iterable<RideRequest> rideRequests = rideRequestRepository.findAllByUserid(userid);
+        String body = mapper.writeValueAsString(rideRequests);
+        return ResponseEntity.ok().body(body);
+    }
+
+    @ApiOperation(value = "Create a new ride request")
+    @PreAuthorize("hasRole('ROLE_RIDER')")
+    @PostMapping("/post")
+    public RideRequest postRideRequest(
+        @ApiParam(name="userid", type="Long", value="userid of the rider", example="123", required=true)  @RequestParam Long userid,
+        @ApiParam(name="day", type="String", value="day, one of [Monday, Tuesday,..., Sunday]", example="Monday", required=true) @RequestParam String day,
+        @ApiParam(name="fullName", type="String", value="name of the rider", example="Joe", required=true) @RequestParam String fullName,
+        @ApiParam(name="course", type="String", value="course to attend", example="CS 156", required=true) @RequestParam String course,
+        @ApiParam(name="startTime", type="String", value="start time, format: HH:MM XM", example="12:00 AM", required=true) @RequestParam String startTime,
+        @ApiParam(name="stopTime", type="String", value="stop time, format: HH:MM XM", example="12:00 AM", required=true) @RequestParam String stopTime,
+        @ApiParam(name="building", type="String", value="destination building", example="Phelps", required=true) @RequestParam String building,
+        @ApiParam(name="room", type="String", value="destination room", example="EUCR, 3525", required=true) @RequestParam String room,
+        @ApiParam(name="pickup", type="String", value="pickup location", example="San Rafael Hall", required=true) @RequestParam String pickup
+        )
+        throws JsonProcessingException {
+
+        RideRequest rideRequest = new RideRequest();
+        rideRequest.setUserid(userid);
+        rideRequest.setDay(day);
+        rideRequest.setFullName(fullName);
+        rideRequest.setCourse(course);
+        rideRequest.setStartTime(startTime);
+        rideRequest.setStopTime(stopTime);
+        rideRequest.setBuilding(building);
+        rideRequest.setRoom(room);
+        rideRequest.setPickup(pickup);
+
+        RideRequest saved = rideRequestRepository.save(rideRequest);
+
+        return saved;
+    }
+
+    @ApiOperation(value = "Delete a ride request")
+    @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_RIDER')")
+    @DeleteMapping("/delete")
+    public Object deleteRideRequest_Admin(
+            @ApiParam("id") @RequestParam Long id) {
+              RideRequest rideRequest = rideRequestRepository.findById(id)
```

Comment:
This should be separated into two methods; one for the ROLE_ADMIN, and one for the ROLE_DRIVER.

The one for the admin works just like you have it here. 

The one for the DRIVER should check whether the user id of the ride the user has requested to delete matches the user's id, and if not, it should throw an exception.  Again, you can find the relevant code in proj-courses and probably also proj-happycows.  I'll link you to it.

Reply:
Another option is to keep this as one method, but then check whether the user has the admin role inside the method, and skip the  check "does the user_id on the ride request match the current user's id" in cases where the user is an admin.